### ISSUE TITLE
Hash key confusion - using strings and symbols to key config hash

### ISSUE THREAD

**Issue by justenwalker** [puppetlabs/r10k#18]
I get an error when I try to run r10k synchronize.  It appears that R10K::Root wants
its config hash to be keyed on symbols, but its being given in strings instead.  

As you can see, it throws a runtime error even though ref is defined:

```
Runtime error: #<RuntimeError: Unable to resolve directory name from options {"remote"=>"git@git.example.com:puppet-modules.git", "ref"=>"production", :basedir=>"/etc/puppet/environments"}>
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/root.rb:71:in `parse_initialize_hash'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/root.rb:26:in `initialize'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:67:in `new'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:67:in `load_all'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:66:in `each'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:66:in `load_all'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:57:in `each_pair'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:57:in `load_all'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment/environment_collection.rb:15:in `initialize'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment.rb:40:in `new'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment.rb:40:in `collection'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/deployment.rb:35:in `environments'
/var/lib/gems/1.8/gems/r10k-0.0.10/lib/r10k/cli/synchronize.rb:18:in `command'
/var/lib/gems/1.8/gems/cri-2.3.0/lib/cri/command.rb:296:in `call'
/var/lib/gems/1.8/gems/cri-2.3.0/lib/cri/command.rb:296:in `run_this'
/var/lib/gems/1.8/gems/cri-2.3.0/lib/cri/command.rb:249:in `run'
/var/lib/gems/1.8/gems/cri-2.3.0/lib/cri/command.rb:262:in `run'
/var/lib/gems/1.8/gems/r10k-0.0.10/bin/r10k:7
/usr/local/bin/r10k:19:in `load'
/usr/local/bin/r10k:19
```

As a work-around I've tried to make it tolerant of both strings and hashes, but I'm not sure what is the best idiomatic solution to this problem.

Reference Commit: a0a5189e2a

**Comment by adrienthebo** [comment]
I've actually been bitten by this as well; I'm planning on adding validation to the config file so that if there's a mismatch like this it's reported sooner rather than later.

**Comment by adrienthebo** [comment]
This should be partially resolved in a1836ceeec8468e845667de02f1d56b635540f69; the top level `:sources` key is still sensitive to Symbols vs Strings.

**Comment by adrienthebo** [comment]
Released in 1.0.0rc1
